packages/synthetix-main/contracts/storage/Pool.sol

```diff
@@ -192,24 +198,31 @@ library Pool {
      */
     function getSystemMaxValuePerShare(
         Data storage self,
-        Market.Data storage marketData,
-        uint unusedCreditCapacityD18
+        uint128 marketId, // we are keeping this value as a uint128 for now for testability reasons
+        uint creditCapacityD18,
+        int debtD18
     ) internal view returns (int) {
+        Market.Data storage marketData = Market.load(marketId);
         uint minLiquidityRatioD18 = SystemPoolConfiguration.load().minLiquidityRatioD18;
 
         // Calculate the margin of debt that the market could incur to hit the system wide limit.
-        uint marginD18;
+        uint totalSharesD18 = self.vaultsDebtDistribution.totalSharesD18;
+        int valuePerShare = marketData.poolsDebtDistribution.getValuePerShare();
```

Comment:
Add Decimal precision to variable name (valuePerShare)

Reply:
fixed